src/atlases.jl

```diff
@@ -0,0 +1,242 @@
+using retractions, inverse retractions and coordinate calculation for a given basis.
+
+# See also
+
+[`AbstractAtlas`](@ref), [`AbstractInverseRetractionMethod`](@ref),
+[`AbstractRetractionMethod`](@ref), [`AbstractBasis`](@ref)
+"""
+struct RetractionAtlas{
+    TInvRetr<:AbstractInverseRetractionMethod,
+    TRetr<:AbstractRetractionMethod,
+    TBasis<:AbstractBasis{<:Any,TangentSpaceType},
+} <: AbstractAtlas
+    invretr::TInvRetr
+    retr::TRetr
+    basis::TBasis
+end
+
+function RetractionAtlas(
+    invretr::AbstractInverseRetractionMethod,
+    retr::AbstractRetractionMethod,
+)
+    return RetractionAtlas(invretr, retr, DefaultOrthonormalBasis())
+end
+RetractionAtlas() = RetractionAtlas(LogarithmicInverseRetraction(), ExponentialRetraction())
+
+get_default_atlas(M::Manifold) = RetractionAtlas()
+
+"""
+    get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
+
+Calculate coordinates of point `p` on manifold `M` in chart from an [`AbstractAtlas`](@ref)
+`A` at index `i`.
+"""
+get_point_coordinates(::Manifold, ::AbstractAtlas, ::Any, ::Any)
+
+function get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
+    x = allocate_result(M, get_point_coordinates, p)
+    get_point_coordinates!(M, x, A, i, p)
+    return x
+end
+
+function allocate_result(M::Manifold, f::typeof(get_point_coordinates), p)
+    T = allocate_result_type(M, f, (p,))
+    return allocate(p, T, manifold_dimension(M))
+end
+
+function get_point_coordinates!(M::Manifold, x, A::RetractionAtlas, i, p)
+    return get_coordinates!(M, x, i, inverse_retract(M, i, p, A.invretr), A.basis)
+end
+
+function get_point_coordinates(M::Manifold, A::RetractionAtlas, i, p)
+    return get_coordinates(M, i, inverse_retract(M, i, p, A.invretr), A.basis)
+end
+
+"""
+    get_point(M::Manifold, A::AbstractAtlas, i, x)
+
+Calculate point at coordinates `x` on manifold `M` in chart from an [`AbstractAtlas`](@ref)
+`A` at index `i`.
+"""
+get_point(::Manifold, ::AbstractAtlas, ::Any, ::Any)
+
+function get_point(M::Manifold, A::AbstractAtlas, i, x)
+    p = allocate_result(M, get_point, x)
+    get_point!(M, p, A, i, x)
+    return p
+end
+
+function allocate_result(M::Manifold, f::typeof(get_point), x)
+    T = allocate_result_type(M, f, (x,))
+    return allocate(x, T, representation_size(M)...)
+end
+
+function get_point(M::Manifold, A::RetractionAtlas, i, x)
+    return retract(M, i, get_vector(M, i, x, A.basis), A.retr)
+end
+
+function get_point!(M::Manifold, p, A::RetractionAtlas, i, x)
+    return retract!(M, p, i, get_vector(M, i, x, A.basis), A.retr)
+end
+
+"""
+    get_chart_index(M::Manifold, A::AbstractAtlas, p)
+
+Select a chart from an [`AbstractAtlas`](@ref) `A` for manifold `M` that is suitable for
+representing neighborhood of point `p`.
+"""
+get_chart_index(::Manifold, ::AbstractAtlas, ::Any)
+
+get_chart_index(::Manifold, ::RetractionAtlas, p) = p
+
+"""
+    transition_map(M::Manifold, A_from::AbstractAtlas, i_from, A_to::AbstractAtlas, i_to, x)
```

Comment:
One could use `i` and `j` for the indices and maybe `A1` and `A2` or `A` and `B` for the atlases (though `B` might be confused with a basis maybe)?

Reply:
I think in this case more descriptive argument names are useful to be able to easily tell which chart is which (is it from 1 to 2 or from 2 to 1?)

Reply:
ok.